### DISCUSSION TITLE
Ability to treat minor version bumps as breaking changes

### DISCUSSION THREAD

**Discussion by mwgamble** [renovatebot/renovate#18562] [Suggest an Idea]
### What would you like Renovate to be able to do?

I'd like Renovate to not recommend a Python version bump from `3.10.8` to `3.11.0` as a minor update that can be merged together with other updates. Python updates of this nature always contain at least some breaking changes, and usually require accompanying changes to a repository that are difficult or unreasonable to automate with Renovate. An example is updating the list of target versions used by the Python code formatter `black`.

However I'd like this functionality to be generic. At work I deal with another piece of software named [Velero](https://github.com/vmware-tanzu/velero) that also doesn't utilise semantic versioning. Each "minor" version bump requires us to perform a manual migration. Ideally, we'd have Renovate understand that "minor" version bumps for this package should be treated as "major" and not automatically merged.

### If you have any ideas on how this should be implemented, please tell us here.

I believe this functionality can be achieved with a combination of settings in a package rule. One idea I had was to use something along the lines of regex versioning of `extractVersion` (I think?) to combine the major and minor versions, so that Renovate would see the version as (for example) "310.8" or "311.0" for the purposes of comparison.

However, I can envisage this being difficult to implement when trying to combine it with other settings. It would also get more complicated if you wanted to provide these rules as presets, and could result in lots of configuration duplication depending on what individual users want to achieve.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
Renovate by default does not group together minor updates. If you have added a rule to group non-major updates, you can exclude packages such as `python` from them.
